src/legacy/core_plugins/vis_type_vislib/public/vislib/helpers/hierarchical/build_hierarchical_data.ts

```diff
@@ -18,37 +18,66 @@
  */
 
 import { toArray } from 'lodash';
-import { getFormat } from 'ui/visualize/loader/pipeline_helpers/utilities';
+import { getFormatService } from '../../../services';
+import { Table } from '../../types';
 
-export const buildHierarchicalData = (table, { metric, buckets = [] }) => {
-  let slices;
-  const names = {};
+export interface Dimension {
+  accessor: number;
+  format: {
+    id?: string;
+    params?: { pattern?: string; [key: string]: any };
+  };
+}
+
+export interface Dimensions {
```

Comment:
Here I based on [this ](https://github.com/elastic/kibana/blob/master/src/plugins/visualizations/public/legacy/build_pipeline.ts#L432)config structure.